kvserver: misc test flake fixes

tbg opened the pull request:
See individual commits.

tbg committed:
split: bump test timeout

Fixes #102512.

Epic: none
Release note: None

tbg committed:
intentresolver: bump test timeout

Fixes #101990.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/102870)
<!-- Reviewable:end -->

tbg committed:
kvserver: deflake TestNodeLivenessRetryAmbiguousResultError

I don't see a reason why there would be exactly two injected
errors, since we don't stop the heartbeat loop in this test.

We expect *at least* two.

Fixes #101722.

Epic: none
Release note: None

kvoli reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/102870#-:-NUwXc04EsDweybpF3uR:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @pavelkalinnikov)



<!-- Sent from Reviewable.io -->

tbg commented:
TFTR!

bors r=kvoli

irfansharif commented:
blathers backport 23.1
